schnorr_fun/src/musig.rs

```diff
@@ -209,11 +209,13 @@ impl KeyList {
     /// unusual case that the tweak is exactly equal to the negation of the aggregated secret key
     /// it returns `None`.
     pub fn tweak(&self, tweak: Scalar<impl Secrecy, impl ZeroChoice>) -> Option<Self> {
-        let mut tweak = s!(self.tweak + tweak).mark::<Public>();
-        let (agg_key, needs_negation) = g!(self.agg_key + tweak * G)
+        let new_tweak = s!(0 + tweak).mark::<Public>();
```

Comment:
what are you trying to achieve here? clone?

Reply:
Hacky way of avoiding the `ZeroChoice` in `g!()` which needs `DecideZero` trait. What is the proper way?

Reply:
What's the error you're getting?

Reply:
If tweak is ZeroChoice:
```
error[E0277]: the trait bound `impl ZeroChoice: DecideZero<_>` is not satisfied
   --> schnorr_fun/src/musig.rs:213:41
    |
213 |         let (agg_key, needs_negation) = g!(self.agg_key + tweak * G)
    |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `DecideZero<_>` is not implemented for `impl ZeroChoice`
    |
note: required by a bound in `scalar_mul_point`
   --> /home/nick/repos/secp256kfun/secp256kfun/src/op.rs:49:9
    |
49  |     Z1: DecideZero<Z2>,
    |         ^^^^^^^^^^^^^^ required by this bound in `scalar_mul_point`
```
I'm currently forcing new_tweak to not be zero choice with `s!(0 + tweak).mark::<Public>();`